### PR TITLE
Redo APIs and simplify specs

### LINKED ISSUE #25: Simplify specs

### LINKED ISSUE #26: Redo the APIs
* Document which formats are based on `Bytes` and `IO`.
* Remove the `String` API.
* ~~[Switch to explicit named arguments or named methods](https://github.com/phenopolis/pluto/pull/23).~~

### PR DESCRIPTION
**Pull request by sanks64**
Closes #25
Closes #26

### CODE REVIEW

File: spec/pluto/format/jpeg_spec.cr [hunk 1]
```diff
@@ -1,49 +1,25 @@
...
-      image = Pluto::RGBAImage.from_jpeg(data)
-
-      Digest::SHA1.hexdigest(image.to_jpeg).should eq "60b7ab88c98807171df33b9242043d1e082b9e1a"
-    end
-
+  describe ".from_jpeg and #to_jpeg" do
     it "works with GrayscaleImage" do
-      data = SpecHelper.pluto_jpg
-      image = Pluto::GrayscaleImage.from_jpeg(data)
+      with_sample("pluto.jpg") do |io|
```
Comment (Vici37) [inline spec/pluto/format/jpeg_spec.cr:6]: Oh! I like it!

File: spec/pluto/operation/bilinear_resize_spec.cr [hunk 2]
```diff
@@ -2,54 +2,46 @@ require "../../spec_helper"
...
+      image = grayscale_sample
+      downsized_image = image.bilinear_resize(480, 360)
+      upsized_image = image.bilinear_resize(800, 600)
 
-      Digest::SHA1.hexdigest(original_image.to_ppm).should eq "d7fa6faf6eec5350f8de8b41f478bf7e8d217fa9"
-      Digest::SHA1.hexdigest(downsized_image.to_ppm).should eq "7a18aea5a8a33fbb74cd12182172fd266f8b9c60"
-      Digest::SHA1.hexdigest(upsized_image.to_ppm).should eq "4091684fe7b44c6d9a61ff732ab8d6f26b129e88"
+      expect_digest image, "1a4d4e43e17f3245cefe5dd2c002fb85de079ae8"
+      expect_digest downsized_image, "e99a957526b32dfbfabff4b580335944a1659b67"
+      expect_digest upsized_image, "9fe83c54452750e436da32b42a9cad5e8a904c1b"
```
Comment (Vici37) [inline spec/pluto/operation/bilinear_resize_spec.cr:12]: Why did these digests change?

Comment (sanks64) [reply spec/pluto/operation/bilinear_resize_spec.cr:12]: Yep. They are sorted alphabetically in file explorers, so I thought it is handy to have them sorted here as well.

File: src/pluto/image.cr [hunk 3]
```diff
@@ -26,6 +27,7 @@ abstract class Pluto::Image
     {% end %}
   end
 
+  # TODO: Remove.
   forward_to_rgb_image from_ppm, from_jpg
```
Comment (Vici37) [inline src/pluto/image.cr:31]: Excellent

**Review by Vici37 — DISMISSED** [review]
Great changes! Regarding the comments for memory consumption in the ppm format - am I correct in thinking that since we intend to read the entire image into memory anyways (in the form of an `Image` object of some sort), the `IO` and `Bytes` methods are the same memory perf wise?

**Comment by sanks64** [comment]
Makes sense. I will delete the comments for now, but we should benchmarks this.

**Pull request merged**